src/components/FormProduct/index.jsx

```diff
@@ -10,38 +14,40 @@ export const FormProduct = () => {
 
   const [product, setProduct] = useState(initialStateProduct);
 
-  const [error, setError] = useState(false);
-
+  const [error, setError] = useState('');
   const handleInputProduct = (e) => {
     setProduct({
       ...product,
       [e.target.name]: e.target.value,
     });
+    setError(
+      e.target.value ? (
+        ''
+      ) : (
+        <span aria-live="polite">All fields are required</span>
+      ),
+    ); //Required field. Immediate validation
   };
-
   const handleSubmitProduct = (e) => {
     e.preventDefault();
-    let { item, nextPurchase, lastPurchasedDate } = product;
+    const { item, nextPurchase } = product;
 
-    if (item.trim() === '' || nextPurchase === 0) {
-      setError(true);
-      return;
-    }
+    //duplication validation
 
-    setError(false);
+    if (productExists(products, item))
```

Comment:
This comment is a bit redundant, and redundant comments should be avoided if possible.

What I think that you could do it to rename `productExists` to `isProductDuplicated` and then remove the comment above it. If you do that, then the function name will imply that we're checking for duplicates.